File: testdata/jennies/builders/initialization_safeguards/PythonBuilder/builders/initialization_safeguards.py
import typing
from ..cog import builder as cogbuilder
from ..models import initialization_safeguards


class SomePanel(cogbuilder.Builder[initialization_safeguards.SomePanel]):
    _internal: initialization_safeguards.SomePanel

    def __init__(self):
        self._internal = initialization_safeguards.SomePanel()

    def build(self) -> initialization_safeguards.SomePanel:
        """
        Builds the object.
        """
        return self._internal    
    
    def title(self, title: str) -> typing.Self:    
        self._internal.title = title
    
        return self
    
    def show_legend(self, show: bool) -> typing.Self:    
        if self._internal.options is None:
            self._internal.options = initialization_safeguards.Options()
        assert isinstance(self._internal.options, initialization_safeguards.Options)
        if self._internal.options.legend is None:
            self._internal.options.legend = initialization_safeguards.LegendOptions()
        assert isinstance(self._internal.options.legend, initialization_safeguards.LegendOptions)
        self._internal.options.legend.show = show
    
        return self
    

File: testdata/jennies/builders/initialization_safeguards/RustBuilder/src/builders/initialization_safeguards.rs
use crate::cog;
use crate::types::initialization_safeguards;

#[derive(Debug, Clone)]
pub struct SomePanelBuilder {
    internal: initialization_safeguards::SomePanel,
    pub(crate) errors: Vec<cog::BuildError>,
}

impl SomePanelBuilder {
    pub fn new() -> Self {
        Self {
            internal: initialization_safeguards::SomePanel::default(),
            errors: Vec::new(),
        }
    }
}

impl Default for SomePanelBuilder {
    fn default() -> Self {
        Self::new()
    }
}

impl SomePanelBuilder {
    pub fn title(mut self, title: String) -> Self {
        self.internal.title = title;

        self
    }
}

impl SomePanelBuilder {
    pub fn show_legend(mut self, show: bool) -> Self {
        self.internal
            .options
            .get_or_insert_with(initialization_safeguards::Options::default)
            .legend
            .show = show;

        self
    }
}

impl cog::Builder<initialization_safeguards::SomePanel> for SomePanelBuilder {
    fn build(&self) -> Result<initialization_safeguards::SomePanel, Vec<cog::BuildError>> {
        if !self.errors.is_empty() {
            return Err(self.errors.clone());
        }

        Ok(self.internal.clone())
    }
}
